### PR TITLE
LPS-173704 Add new CriteriaRowReadable component

### PR DESCRIPTION
**Pull request by claraizquierdo**
This is part of a bigger PR: https://github.com/liferay-tango/liferay-portal/pull/3202

Motivation
=======
Add new component for rendering the criteria in view mode and add the event criteria in view mode.
[Link to story or ticket](https://issues.liferay.com/browse/LPS-173704)

Proposed Solution
========
We have added a new component with all the logic for rendering the criteria in view mode. That component is still not visible in the portal. It will be used by CriteriaRow component in the future.

Steps to Verify:
----------------
Go to people -> segments and create a new segment
Add different types of properties and save
View the new created segment and check that all the criteria are correct
Edit the new created segment and check that all the criteria can be edited

### CODE REVIEW

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:stop

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by claraizquierdo** [comment]
ci:test:tango-acceptance

**Comment by claraizquierdo** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
> ci:test:relevant

Hey @claraizquierdo !!

You are running the test without any change in your side right? If the problems are not related with your PR, please don't run them again, we will send it directly to Brian.

So the CI doesn't collapse for unecessary retries.

Thanks!!

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/constants.es.js [hunk 1]
```diff
@@ -30,6 +32,11 @@ export const NOT_OPERATORS = {
 	NOT_EQ: 'not-eq',
 };
 
+export const NOT_OPERATORS2 = {
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/constants.es.js:35]: Maybe a more descriptive name?

Comment (cgoncas) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/constants.es.js:35]: HAS_OPERATORS?

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/CriteriaRowReadable.es.js [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+	operatorName: 'gt',
+	operatorNot: true,
+	propertyName: 'downloadedDocuments',
+	propertyType: 'count',
+
+	query: {
+		conjunctionName: 'and',
+		groupId: 'group_2',
+		items: [
+			{
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/test/js/components/criteria_builder/CriteriaRowReadable.es.js:122]: With the changes proposed in https://issues.liferay.com/browse/LPS-171785 the right JSON will be something like:

```
    "groupId": "group_0",
    "items": [
      {
        "propertyName": "pageViewed",
        "assetId": "545188693724480037'",
        "day": {
          "value": "2024-01-26T23:00:00.000Z",
          "operator": "eq"
        },
        "operatorName": "ge",
        "value": "1"
      }
    ],
    "conjunctionName": "and"
  }```

**Review by cgoncas — CHANGES_REQUESTED** [review]
Hey @claraizquierdo good job 

I have added some comments because yesterday we changed the format for the events segments criteria, so maybe you need to change some code, at least the tests doesn't look fine.

**Review by Mikellorza — APPROVED** [review]
Hey @claraizquierdo,

I have tested and It seems that everything is working fine. Only I saw an small details but I think that it is not related to this PR and I am going to explain it in another comment.

Thanks

**Comment by Mikellorza** [comment]
Hey @claraizquierdo ,

When I was testing this PR I saw a strange behavior that I am going to try to explain: You have a created segment with two different types of criterias (User and Organization), and between them you have an OR conjuntion. When you add another type of criteria, for example, one of Context criterias, the ui add an AND conjuntion but you have an OR conjuntion within the previous ones, and you can save like you see the segment. This is happening hen you have a segment created and you are trying to add another one type of criterias and is not happening when you try to add a new segment with different types of criterias.

I think that it is an incorrect behavior but maybe I am wrong.

I attach a video explaining the behavior:


https://user-images.githubusercontent.com/46810560/218096969-6df3a129-3651-4500-ac6a-681ff9173c72.mp4


Thanks in advance

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/constants.es.js (hunk 1 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/utils/constants.es.js:35]: Totally agree @cgoncas , I didn't have good ideas for the naming here 😅

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/CriteriaRowReadable.es.js (hunk 2 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/test/js/components/criteria_builder/CriteriaRowReadable.es.js:122]: Yeah, I'm going to adapt the code to those changes

**Comment by claraizquierdo** [comment]
> Hey @claraizquierdo ,
> 
> When I was testing this PR I saw a strange behavior that I am going to try to explain: You have a created segment with two different types of criterias (User and Organization), and between them you have an OR conjuntion. When you add another type of criteria, for example, one of Context criterias, the ui add an AND conjuntion but you have an OR conjuntion within the previous ones, and you can save like you see the segment. This is happening hen you have a segment created and you are trying to add another one type of criterias and is not happening when you try to add a new segment with different types of criterias.
> 
> I think that it is an incorrect behavior but maybe I am wrong.
> 
> I attach a video explaining the behavior:
> 
>  Screenshare.-.2023-02-10.1_47_03.PM.mp4 
> Thanks in advance

Hi @Mikellorza , thank you for your review :) To me this behavior makes sense since the default operator between groups is the AND, but I might be wrong. Anyway, I did not touch anything related to that in this PR so I think this is happening in master too.

**Comment by Mikellorza** [comment]
> Hey @claraizquierdo ,
> 
> When I was testing this PR I saw a strange behavior that I am going to try to explain: You have a created segment with two different types of criterias (User and Organization), and between them you have an OR conjuntion. When you add another type of criteria, for example, one of Context criterias, the ui add an AND conjuntion but you have an OR conjuntion within the previous ones, and you can save like you see the segment. This is happening hen you have a segment created and you are trying to add another one type of criterias and is not happening when you try to add a new segment with different types of criterias.
> 
> I think that it is an incorrect behavior but maybe I am wrong.
> 
> I attach a video explaining the behavior:
> 
>  Screenshare.-.2023-02-10.1_47_03.PM.mp4 
> Thanks in advance

Hey @CarlosBrichete ,

I am not sure if the behavior that I explained in the quoted message is a correct behavior or maybe it is a bug. What do you think? 

Thanks

**Comment by CarlosBrichete** [comment]
Hey @Mikellorza, I'm not sure, maybe it is not a bug but the behaviour is a little bit confusing. However, this is happening in master too, so not related with this PR. Can you create a bug ticket with this behaviour in JIRA so we can investigate furthermore? Thanks!

**Comment by claraizquierdo** [comment]
ci:test:sf

**Comment by claraizquierdo** [comment]
ci:test:relevant

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {PropTypes} from 'prop-types';
+import React, {Component} from 'react';
+
+import {
+	DATE_OPERATORS,
+	NOT_OPERATORS2,
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Should we rename it HAS_OPERATORS?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+	_renderEventCriteriaString = ({criterion, propertyLabel}) => {
+		const notOperators = getSupportedOperatorsFromEvent(
+			SUPPORTED_EVENT_OPERATORS,
+			SUPPORTED_PROPERTY_TYPES,
+			'NOT'
+		);
+
+		const notOperatorKey = criterion.operatorNot
+			? NOT_OPERATORS2.NOT_HAS
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Should we rename it HAS_OPERATORS?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+			: NOT_OPERATORS2.HAS;
```
Comment (cgoncas) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Should we rename it HAS_OPERATORS?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js (hunk 3 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: That change is in the following commit. I can edit the commit to include it in this one.

Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Moved to the correct commit

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js (hunk 4 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Same here

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js (hunk 5 above)
Comment (claraizquierdo) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRowReadable.es.js]: Same here

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Review by cgoncas — APPROVED** [review]
LGTM. @claraizquierdo you don't need to include the @claraizquierdo
[LPS-173704 BuildLang](https://github.com/liferay-tango/liferay-portal/pull/3234/commits/94a7f32738ea3891409d55ec1df0f806435af21f) commit, it usually generate conflicts, and Brian will run it himself.

**Comment by claraizquierdo** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**